shared-components/tl-tournament-card.js

```diff
@@ -251,26 +208,51 @@ export class TournamentCard extends LitElement {
                 return html` <h4>loading</h4> `;
             case State.LOADED:
                 return html`
-                    <h4>Title: ${this.tournament.name}</h4>
+                    <img src=${this.tournament.eventImage} />
+                    <h4 class="event-title">${this.tournament.name}</h4>
                     <!-- FIXME: change the value of the substring to match official given value -->
-                    <div class="paragraph-large">Short description: ${this.tournament.description.substring(0, 30)}</div>
-                    <!-- FIXME: show dates like a countdown -->
-                    <div class="paragraph-medium">Event dates: ${this._formatDates()}</div>
-                    <!-- FIXME: show real prize amount -->
-                    <!-- FIXME: use the localization API to generate the plural of the currency symbol -->
-                    <div class="paragraph-large">Prize money: ${this.tournament.fee}&nbsp;${this.tournament.currency.symbol}s</div>
-                    <mwc-button class="details-button button-label-large" label="Details" outlined @click="${(event) => this._handleDetailsButton(event)}"></mwc-button>
+                    <div class="paragraph-large event-text">
+                        ${this.tournament.description.substring(0, 30)}
+                    </div>
+                    <tl-tournament-countdown
+                        class="counter"
+                        date-start="${this.tournament.dateStart}"
+                        date-end="${this.tournament.dateEnd}"
+                    ></tl-tournament-countdown>
+                    <div class="paragraph-large event-prize">
+                        Prize money <span class="event-prize-amount">${this._currentCurrency}</span>
+                    </div>
+                    <mwc-button
+                        class="details-button button-label-large"
+                        label="Details"
+                        outlined
+                        @click="${(event) => this._handleDetailsButton(event)}"
+                    ></mwc-button>
                 `;
             case State.ERROR:
                 return html` <h4>error</h4> `;
         }
     }
 
-    willUpdate() {
+    async willUpdate() {
```

Comment:
First `willUpdate()` is not expected to return a Promise, and in fact should not block the rendering lifecycle, otherwise the `lit` stack will stop refreshing the widget :(

And look, you don't even have an `await` in that function...